pkg/plugin/plugin.go

```diff
@@ -280,10 +319,30 @@ func CmdAdd(args *skel.CmdArgs) error {
 	}
 
 	var vlanTagNum uint = 0
+	var vlanMode = "access"
 	trunks := make([]uint, 0)
-	portType := "access"
-	if netconf.VlanTag == nil || len(netconf.Trunk) > 0 {
-		portType = "trunk"
+	cvlans := make([]uint, 0)
+
+	if netconf.VlanMode != nil {
+		vlanMode = *netconf.VlanMode
+	}
+
+	if netconf.VlanTag != nil && netconf.Cvlan != nil {
```

Comment:
What if cvlan is not specified? It should still configure qinq, just without any limitation for cvlans (if none is set, OVS should pass all)